src/main/scala/is/hail/sparkextras/ContextRDD.scala

```diff
@@ -157,6 +157,15 @@ class ContextRDD[C <: AutoCloseable, T: ClassTag](
     onRDD(rdd => new AdjustedPartitionsRDD(rdd, contextIgnorantAdjustments))
   }
 
+  def coalesce(numPartitions: Int, shuffle: Boolean = false): ContextRDD[C, T] =
+    // NB: the run marks the end of a context lifetime, the next one starts
+    // after the coalesce/shuffle
```

Comment:
Is coalesce with `shuffle = false` really the end of a context lifetime?

Reply:
I think you're right, there's no network traffic if shuffle = false...